### ISSUE TITLE
Sidekiq logging millions of errors: "uninitialized constant Maintenance" 

### ISSUE THREAD

**Issue by zedrdave** [mastodon/mastodon#20250]
### Steps to reproduce the problem

1. Install Mastodon from source
2. upgrade from rc1 to rc2
3. …

### Expected behaviour

Mastodon runs normally

### Actual behaviour

Sidekiq is spitting non-stop warnings into syslog: filling up the whole server in a matter of hours and crashing

### Detailed description

Not entirely sure at what point the problem started (only realised it when inquiring why syslog had ballooned up to 15GB).

Rebooting, cleaning cache, re-running rake db:migrate, restarting the service: no luck. Each time I restart the service, I get dozens of warnings per second.

I guess I could disable Warning-level logging, but that does not seem like a great idea.

```
Nov  9 16:33:44 ganbare bundle[5895]: 2022-11-09T15:33:44.191Z pid=5895 tid=2qar WARN: NameError: uninitialized constant Maintenance
Nov  9 16:33:44 ganbare bundle[5895]: 2022-11-09T15:33:44.192Z pid=5895 tid=2qar WARN: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/processor.rb:282:in `const_get'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/processor.rb:282:in `block in constantize'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/processor.rb:279:in `each'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/processor.rb:279:in `inject'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/processor.rb:279:in `constantize'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/processor.rb:132:in `block (5 levels) in dispatch'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/rails.rb:14:in `block in call'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/activesupport-6.1.7/lib/active_support/execution_wrapper.rb:91:in `wrap'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/activesupport-6.1.7/lib/active_support/reloader.rb:72:in `block in wrap'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/activesupport-6.1.7/lib/active_support/execution_wrapper.rb:91:in `wrap'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/activesupport-6.1.7/lib/active_support/reloader.rb:71:in `wrap'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/rails.rb:13:in `call'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/processor.rb:131:in `block (4 levels) in dispatch'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/processor.rb:263:in `stats'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/processor.rb:126:in `block (3 levels) in dispatch'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/job_logger.rb:13:in `call'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/processor.rb:125:in `block (2 levels) in dispatch'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/job_retry.rb:80:in `global'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/processor.rb:124:in `block in dispatch'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/job_logger.rb:39:in `prepare'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/processor.rb:123:in `dispatch'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/processor.rb:168:in `process'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/processor.rb:78:in `process_one'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/processor.rb:68:in `run'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/component.rb:8:in `watchdog'
Nov  9 16:33:44 ganbare bundle[5895]: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/sidekiq-6.5.7/lib/sidekiq/component.rb:17:in `block in safe_thread'
Nov  9 16:33:44 ganbare bundle[5895]: 2022-11-09T15:33:43.547Z pid=5895 tid=2qgv WARN: {"context":"Job raised exception","job":{"retry":true,"queue":"pull","class":"Maintenance::UncacheMediaWorker","args":[4408],"jid":"7e709f2724dcbea2cf5c20b0","created_at":1573398008.872565,"enqueued_at":1668005885.431912,"error_message":"uninitialized constant Maintenance","error_class":"NameError","failed_at":1667774082.7567837,"retry_count":15,"retried_at":1667942379.141511}}
Nov  9 16:33:44 ganbare bundle[5895]: 2022-11-09T15:33:44.192Z pid=5895 tid=2qgv WARN: NameError: uninitialized constant Maintenance
```

### Specifications

Mastodon: 4.0.0rc2
Ruby: ruby 2.7.4p191 (2021-07-07 revision a21a3b7d23) [x86_64-linux-gnu]
Node: v16.18.1

**Comment by matthewford** [comment]
It seems like these jobs were removed or renamed between RCs are you sure sidekiq has been restarted?

**Comment by zedrdave** [comment]
I restarted everything and even rebooted the server… Nothing seemed to address the problem.

Reinstalling the whole instance seems to have fixed it.

I am not entirely sure what was the root cause, but it seems like there should be some stopgap to prevent the possibility of sidekiq filling up 20GB of logs…?

**Comment by matthewford** [comment]
Sidekiq should be stopped *after* the jobs in the current queue have completed when you're removing a worker (which they have done - the error is that Sidekiq is trying to load a file that no longer exists); if that's not mentioned in the upgrade guide, then it should be added. Alternatively, you should drop the maintenance job queue before hand.

**Comment by matthewford** [comment]
There is also a sidekiq option to not store the backtrace (and reduce your log output sidekiq_options backtrace: false)

**Comment by zedrdave** [comment]
Honestly, the amount logged probably wouldn't matter all that much ultimately: even if it filled up at a tenth of the speed, it would still quickly overflow.

Thanks for the tips: I'll keep that in mind for next time.

I am not well-versed enough in Sidekiq management to clearly understand what happened, and the problem was resolved by nuking everything, but some info in the documentation regarding what to do when similar issues happen, might be useful (I don't think I saw anything pertaining to this anywhere).
